aws/oidc_provider_test.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/iam"
+	"github.com/gruntwork-io/cloud-nuke/config"
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// NOTE: OpenID Connect Provider is a global resource, so we use the default region for the tests.
+
+func TestListOIDCProviders(t *testing.T) {
+	t.Parallel()
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(defaultRegion)})
+	require.NoError(t, err)
+	svc := iam.New(session)
+
+	oidcProviderARN := createOIDCProvider(t, svc, "base", defaultRegion)
+	defer deleteOIDCProvider(t, svc, oidcProviderARN, true)
+
+	providerARNs, err := getAllOIDCProviders(session, time.Now(), config.Config{})
+	require.NoError(t, err)
+	assert.Contains(t, aws.StringValueSlice(providerARNs), aws.StringValue(oidcProviderARN))
+}
+
+func TestTimeFilterExclusionNewlyCreatedOIDCProvider(t *testing.T) {
+	t.Parallel()
+
+	session, err := session.NewSession(&aws.Config{Region: aws.String(defaultRegion)})
+	require.NoError(t, err)
+	svc := iam.New(session)
+
+	oidcProviderARN := createOIDCProvider(t, svc, "base", defaultRegion)
+	defer deleteOIDCProvider(t, svc, oidcProviderARN, true)
+
+	// Assert OpenID Connect Provider is picked up without filters
+	providerARNsNewer, err := getAllOIDCProviders(session, time.Now(), config.Config{})
+	require.NoError(t, err)
+	assert.Contains(t, aws.StringValueSlice(providerARNsNewer), aws.StringValue(oidcProviderARN))
+
+	// Assert provider doesn't appear when we look at providers older than 1 Hour
+	olderThan := time.Now().Add(-1 * time.Hour)
+	providerARNsOlder, err := getAllOIDCProviders(session, olderThan, config.Config{})
+	require.NoError(t, err)
+	assert.Contains(t, aws.StringValueSlice(providerARNsOlder), aws.StringValue(oidcProviderARN))
```

Comment:
This was actually a bug in the test, where I was supposed to check `asset.NotContains` instead of `assert.Contains`, since I'm supposed to make sure the provider is excluded when a time filter is introduced.

Reply:
Fixed in 917dd0cc41510bca4c6092e1a7fa3601ac508aea